cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -562,7 +562,12 @@ protected static SqlParts whereSqlForAndClause(List<Object> clause, String index
                         // This was validated during normalization.
```

Comment:
You need to update the comment above this, as it is not limited to String or non-Float now.